### PR TITLE
Do not cast using NGSIv2 types in updateAction (version 2)

### PR DESCRIPTION
**Pull request by AlvaroVega**
Currently there are some automatic cast about types:
- Number
- Text
- Booleam
- None
- DateTime

What do we want?
Currently updateAction in version 2 is trying to cast (after expandVars) to NGSIv2 types like:

```javascript
var changes = {};
    action.parameters.attributes.forEach(function(attr) {
        // Direct value for Text, DateTime, and others. Apply expandVar for strings
        let theValue = myutils.expandVar(attr.value, event, true);
        let theType = myutils.expandVar(attr.type, event);
        // Metadata should be null or object
        let theMeta = attr.metadata;
        var date;
        switch (theType) {
             case 'Text':
                 theValue = theValue.toString();
                 break;
             case 'Number':
                 theValue = parseFloat(theValue);
                 break;
             case 'Boolean':
                 if (typeof theValue === 'string') {
                     theValue = theValue.toLowerCase().trim() === 'true';
                 }
                 break;
             case 'DateTime':
                 if (parseInt(theValue).toString() === theValue) {
                     // Parse String with number (timestamp__ts in Perseo events)
                     theValue = parseInt(theValue);
                 }
                 date = new Date(theValue);
                 theValue = isNaN(date.getTime()) ? theValue : date.toISOString();
                 break;
             case 'None':
                 theValue = null;
                 break;
         }
        var key = myutils.expandVar(attr.name, event);
        changes[key] = {
            value: theValue,
            type: theType
        };
```

### CODE REVIEW

**Comment by AlvaroVega** [comment]
@fgalan Which version is used by default (v1 or v2) for that notification?

**Comment by AlvaroVega** [comment]
@fgalan Maybe we can add a configuration flag option to enable or disable it, instead of remove all ?

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- Disable automatic cast for NGSIv2 types
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]:
```suggestion
- Add: castType (PERSEO_CAST_TYPE) to enable NGSIv2 type based casting (default is to use JSON native types in values)
```

File: bin/perseo [hunk 2]
```diff
@@ -157,7 +158,7 @@ function loadConfiguration() {
     config.smpp.enabled = process.env.PERSEO_SMPP_ENABLED || config.smpp.enabled;
     config.endpoint.noticesPath = process.env.PERSEO_NOTICES_PATH || config.endpoint.noticesPath;
     config.endpoint.rulesPath = process.env.PERSEO_RULES_PATH || config.endpoint.rulesPath;
-
+    config.castTypes = process.env.PERSEO_CAST_TYPES || config.castTypes;
```
Comment (fgalan) [inline bin/perseo]: Should the new config parameter / env var to be documented in some .md file? Not remember how we are doing this in Perseo...

Comment (AlvaroVega) [reply bin/perseo]: Done in 66142b24de3cd86bdb113738e69543a32665c7ab

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (fgalan) [reply CHANGES_NEXT_RELEASE]: Fixed in a3104d8

File: documentation/configuration.md [hunk 3]
```diff
@@ -48,6 +48,7 @@ The following table shows the environment variables available for Perseo configu
 | PERSEO_AUTHENTICATION_PORT         | Port of the authentication endpoint                                                        |
 | PERSEO_AUTHENTICATION_USER         | User to perform authentication                                                             |
 | PERSEO_AUTHENTICATION_PASSWORD     | Password for the user to perform authentication                                            |
+| PERSEO_CAST_TYPE                   | Cast or not attribute values to expected type conform NGSIv2 (false by default)            |
```
Comment (fgalan) [inline documentation/configuration.md]:
```suggestion
| PERSEO_CAST_TYPE                   |  If true, enable attribute value casting based in NGSIv2 attribute types if true. If false (default), the JSON native type for the attribute value is used.            |
```

**Pull request merged**